### PR TITLE
Self-sovereign Identity

### PR DESCRIPTION
**Pull request by 3esmit**
# Self-sovereign Identity

## Preamble

    Idea: 145-identity
    Title: Self-sovereign Identity
    Status: Draft
    Created: 2018-05-04
    Requires: 152-friends-recovery ERC725 ERC735
    
## Summary

Creates a common user-owned credentials for authentication on smart contracts and for enhancing users security with multisig and optional recovery.

## Swarm Participants

- Lead Contributor:  @3esmit
- Contributor: @richard-ramos 
- UX (Mist): @alexvandesande 
- Testing & Evaluation:  (help needed)
- PM: (help needed)
- UX (Status):  (help needed)

Related to:
#152 https://github.com/ethereum/EIPs/issues/725 https://github.com/ethereum/EIPs/issues/735

### CODE REVIEW

File: ideas/145-identity.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    Title: Identity
+    Status: In Progress 
+    Created: 2018-05-04
+    Requires: #27 #73 ERC725 ERC735
+    
+## Summary
+
+Creates a common user-owned credentials for authentication on smart contracts and for enhancing users security with multisig and optional recovery.
+
+## Swarm Participants
```
Comment (chadyj) [inline ideas/145-identity.md]: This sounds like a big project. Do you need other technical contributors?
What type of UX and PM support do you need?

Comment (3esmit) [reply ideas/145-identity.md]: Project Manager should help integrate Identity in Status other projects, if any. I could also take the role of PM, but If anyone want to take place.
UX is also about integration of Identity, not sure what to comment about it

File: ideas/145-identity.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+## Success Metrics
+<!-- Assuming the idea ships, what would success look like? What are the most important metrics that you would move? -->
+
+Users opt-in using an Identity to enhance interaction with decetranlize systems (such as Status itself)
+
+<!-- Example: Onboarding conversion rate. Target >30% full funnel -->
+
+## Exit criteria
+<!-- Launch new onboarding UI flow -->
+Community consensus in ERC725 and ERC735 interfaces, launched on Status.
```
[the remark this answers is not in the document]
Comment (3esmit) [reply ideas/145-identity.md]: Ok, I will. Community is the ethereum developers.

File: ideas/145-identity.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+
+- Recovery: Setup/Update/Cancel Update 
+- Request recovery to selected "friends".
+- Confirm/reject other's recovery request
+
+## Success Metrics
+<!-- Assuming the idea ships, what would success look like? What are the most important metrics that you would move? -->
+
+Users opt-in using an Identity to enhance interaction with decetranlize systems (such as Status itself)
```
[the remark this answers is not in the document]
Comment (3esmit) [reply ideas/145-identity.md]: It's about decentralization.

File: ideas/145-identity.md (hunk 1 above)
Comment (3esmit) [reply ideas/145-identity.md]: I will externalize Recovery development in a new idea.

File: ideas/145-identity.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
+# Self-sovereign Identity
+
+## Preamble
+
+    Idea: 145-identity
+    Title: Self-sovereign Identity
+    Status: In Progress 
+    Created: 2018-05-04
+    Requires: 150-gas-abstraction 151-ens-usernames 152-friends-recovery ERC725 ERC735
```
Comment (oskarth) [inline ideas/145-identity.md]: Considering how many dependencies this ideas has, maybe we can make the scope more minimal? So it's less about integrating in app straight away and more about finalizing PoC and influencing ERC725/735 based on desired UX. Then we can have a separate swarm for implementing it in app (and getting it audited) once requirements are done. WDYT?

File: ideas/145-identity.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+## Dates
+
+### Proof-of-concept
+
+Goal Date: 2018-05-15
+
+Description: 
+- Working contracts system 
+- Basic UI for direct interaction with blockchain
```
Comment (oskarth) [inline ideas/145-identity.md]: What does this mean?

File: ideas/145-identity.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    
+## Success Metrics
+
+Status Users opt-in using an Identity to enhance interaction with decetralize systems (such as Status itself). 
+Developers start using Status Identity implementation as base.
+New users adopt Status because Identity support, which will be required for many dapps in future.
+
+## Exit criteria
+
+Fully integrated on Status App.
```
Comment (oskarth) [inline ideas/145-identity.md]: Would be useful to frame this in terms of specific capabilities provided in app

File: ideas/145-identity.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+- Smart contracts: See [GDocs Identity Terms Intent](https://docs.google.com/document/d/1K8tFjGneScuKiudpD3HfSpAd4eOe8012jn0Q3iDxEV0/edit#) and [Identity Readme](https://github.com/status-im/contracts/blob/develop/Identity.md)
+- Basic contracts demo: Helps engaging developers and bug hunting. 
+- Identity Sign-in: Actions for user creating or recovering their wallets.
+- Identity Usage: User may select their Identity to make tansactions in behalf of.
+- Identity Management: User may setup keys & recovery options for its identity, confirm claims and even update the contract code.
+
+### Requirements & Dependencies
+
+- ERC725 and ERC735 should be approved as standard and used in contracts interface, this swarm might influence how this standards end up defined, this is important for the solution working better with unknown future applications.
```
Comment (oskarth) [inline ideas/145-identity.md]: For each of these requirements, it'd be useful to understand at what point they are blockers

File: ideas/145-identity.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
+# Self-sovereign Identity
+
+## Preamble
+
+    Idea: 145-identity
+    Title: Self-sovereign Identity
+    Status: In Progress
```
Comment (oskarth) [inline ideas/145-identity.md]: Seems like idea as a whole is still in draft stage, but PoC is in progress? Not sure how we want to reflect this given all the requirements

File: ideas/145-identity.md (hunk 6 above)
Comment (3esmit) [reply ideas/145-identity.md]: Sure, good point, I'll elaborate on that,

File: ideas/145-identity.md (hunk 8 above)
Comment (3esmit) [reply ideas/145-identity.md]: Yes, the "idea/swarm" is in draft, but the PoC is in progress. 
I'll sort out this.

File: ideas/145-identity.md (hunk 7 above)
Comment (3esmit) [reply ideas/145-identity.md]: Identity itself is independent, the requirements are mostly for better UX and ERCs for developer community.

**Comment by oskarth** [comment]
Cheers, merging

**Pull request merged**